Pod/Classes/VideoComponent/ARVideoManager.m

```diff
@@ -0,0 +1,89 @@
+#import <AVFoundation/AVFoundation.h>
+#import "ARVideoManager.h"
+
+
+@interface ARVideo : UIView
+@property (nonatomic, strong, readwrite) AVPlayer *player;
+@property (nonatomic, strong, readwrite) AVPlayerLayer *playerLayer;
+@end
+
+
+@implementation ARVideo
+
+- (void)setFrame:(CGRect)frame
+{
+    [super setFrame:frame];
+    self.playerLayer.frame = self.bounds;
+}
+
+- (void)playerDidReachEnd:(NSNotification *)notification {
+    AVPlayerItem *playerItem = [notification object];
+    [playerItem seekToTime:kCMTimeZero];
+    [self.player play];
+}
+
+@end
+
+
+@implementation ARVideoManager
+
+RCT_CUSTOM_VIEW_PROPERTY(source, NSDictionary, ARVideo)
+{
+    NSDictionary *source = [RCTConvert NSDictionary:json];
+    bool isNetwork = [RCTConvert BOOL:[source objectForKey:@"isNetwork"]];
+    bool isAsset = [RCTConvert BOOL:[source objectForKey:@"isAsset"]];
+    NSString *uri = [source objectForKey:@"uri"];
+    NSString *type = [source objectForKey:@"type"];
+    NSString *resizeMode = [source objectForKey:@"resizeMode"];
+
+    NSString *videoGravity;
+    if ([resizeMode isEqualToString:@"stretch"]) {
+        videoGravity = AVLayerVideoGravityResize;
+    } else if ([resizeMode isEqualToString:@"contain"]) {
+        videoGravity = AVLayerVideoGravityResizeAspect;
+    } else if ([resizeMode isEqualToString:@"cover"]) {
```

Comment:
We need to use the `cover` aspect ratio, yet this is where things things are centered anchorPoint wise. Need to be left aligned, yet when setting the `view.playerLayer.anchorPoint` it aligns the layer to the left, but the video is still clipped. \

<img width="400" alt="screen shot 2018-09-18 at 6 04 01 pm" src="https://user-images.githubusercontent.com/236943/45727522-f541cd00-bb78-11e8-8023-5bf170b2ac7d.png">

to 

<img width="399" alt="screen shot 2018-09-18 at 6 06 09 pm" src="https://user-images.githubusercontent.com/236943/45727528-fb37ae00-bb78-11e8-88f3-6338f2b47ae3.png">